### PR TITLE
Soft deprecate GenesisConfig

### PR DESCRIPTION
**Pull request by juangirini**
Relates to paritytech/polkadot-sdk#175

This PR Soft deprecates the runtime level `GenesisConfig` by renaming it to `RuntimeGenesisConfig`.
An alias `GenesisConfig` is created to keep backwards compatibility. No deprecation warning has been added at this stage as that will happen later in a separate PR as indicated in paritytech/polkadot-sdk#175

### CODE REVIEW

File: frame/support/procedural/src/construct_runtime/expand/config.rs [hunk 1]
```diff
@@ -74,12 +74,18 @@ pub fn expand_outer_config(
 		#[serde(rename_all = "camelCase")]
 		#[serde(deny_unknown_fields)]
 		#[serde(crate = "__genesis_config_serde_import__")]
-		pub struct GenesisConfig {
+		pub struct RuntimeGenesisConfig {
 			#fields
 		}
 
+		// TODO [#14065]
+		// - deprecate it
```
Comment (ggwpez) [inline frame/support/procedural/src/construct_runtime/expand/config.rs:82]: Could you maybe temporarily put a deprecation here, to see that the CI stays green and we got all cases?

Comment (juangirini) [reply frame/support/procedural/src/construct_runtime/expand/config.rs]: Sure, that's what I did locally. Let me know once you confirm it yourself so I can remove it

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2885859

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2885858

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2886089

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2886091

**Review by bkchr — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2887653

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2887652

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2887927

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2887926

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2888134

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2888135

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2888407

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2888406

**Review by sam0x17 — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

File: frame/support/procedural/src/construct_runtime/expand/config.rs (hunk 1 above)
Comment (ggwpez) [reply frame/support/procedural/src/construct_runtime/expand/config.rs]: Yes you can do them lean without companion (since that is annoying).

**Comment by juangirini** [comment]
bot rebase

**Comment by juangirini** [comment]
bot merge

**Review by muharem — APPROVED** [review]

File: frame/support/procedural/src/construct_runtime/expand/config.rs [hunk 2]
```diff
@@ -74,12 +74,18 @@ pub fn expand_outer_config(
...
+		// - later, remove it
```
Comment (liamaharon) [inline frame/support/procedural/src/construct_runtime/expand/config.rs:83]: Personally I'd refrain from putting deprecation steps themselves in the code comment and just list them in the issue.

The code comment could say something like "GenesisConfig is in the process of being deprecated [issue link]"

**Review by liamaharon — APPROVED** [review]

**Comment by juangirini** [comment]
bot merge

**Pull request merged**